shared/types/User.type.ts

```diff
@@ -0,0 +1,59 @@
+import type { BaseList } from "./Base.type"
+
+/**
+ * Represents a list of users with their ids, usernames, roles, and verification statuses.
+ * Extends the BaseList interface.
+ */
+export interface UserList extends BaseList {
+  /**
+   * The id of the user.
+   */
+  id: string
+  /**
+   * The username of the user.
+   */
+  username: string
+  /**
+   * The role of the user.
+   */
+  role: string
+  /**
+   * Whether the user has verified their account.
+   */
+  verified: string
+}
+
+/**
+ * Represents a user with their id, username, role, email, and verification status.
+ */
+export interface UserDetail extends UserList {
+  /**
+   * The email of the user.
+   */
+  email: string
+}
+
+/**
+ * Represents an update to a user
+ */
+export interface UserUpdate {
```

Comment:
The UserUpdate should only have username and email to change. The role and verified status should be changed via separate endpoints which are not in the scope of this pr